File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        ListNode ret = null;
        for(int i = 0; i < lists.length; i++){
            if(lists[i]!=null){
                if(ret==null){
                    ret = lists[i];
                } else {
                    ret = mergeTwo(ret, lists[i]);
                }
            }
        }
        return ret;
    }

    private ListNode mergeTwo(ListNode a, ListNode b){
        ListNode ret = new ListNode();
        ListNode current = ret;
        while(a!=null && b!=null){
            if(a.val<=b.val){
                current.next = a;
                a=a.next;
            } else {
                current.next = b;
                b=b.next;
            }
            current = current.next;
        }
        if(a==null){
            current.next = b;
        }else {
            current.next = a;
        }
        return ret.next;
    }
}

File: Data Structures & Algorithms/add-two-numbers/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        l1v = self.toNum(l1)
        l2v = self.toNum(l2)
        total = l1v + l2v
        print(l1v, l2v,total)
        head = ListNode()
        cur = head
        while total>0:
            val = total % 10
            total//=10
            node = ListNode(val)
            cur.next = node
            cur = cur.next
        return head.next if head.next else ListNode(0)

    def toNum(self, l:Optional[ListNode]) -> int:
        ret = 0
        mul = 1
        while l:
            val = l.val*mul
            mul*=10
            ret+=val
            l=l.next
        return ret